app/models/vm/operations.rb

```diff
@@ -14,7 +14,14 @@ module Vm::Operations
   end
 
   def cockpit_url
-    URI::HTTP.build(:host => ipaddresses.first, :port => 9090).to_s
+    URI::HTTP.build(:host => get_ipv4_address, :port => 9090).to_s
+  end
+
+  def return_ipv4_address
+    ipaddresses.each do |ip|
```

Comment:
`ipaddresses.find { |ip| IPAddr.new ip).ipv4? } || ipaddresses.first`

would be shorter and more ruby like